HybridCamLib/src/cam/CamView+Gesture.swift

```diff
@@ -6,32 +6,77 @@ extension CamView: UIGestureRecognizerDelegate {
     * Adds gesture event
     */
    @objc open func addGestureRecognizer() {
-      with(UITapGestureRecognizer(target: self, action: #selector(onPreviewViewTap))) {
-         self.previewView.addGestureRecognizer( $0 )
+      //set up tap gesture recognizers
+      let singleTapGestureRecognizer = {
+         let gesture = UITapGestureRecognizer(target: self, action: #selector(onPreviewViewTap))
+         self.previewView.addGestureRecognizer(gesture)
       }
+      with(UITapGestureRecognizer(target: self, action: #selector(onPreviewViewDoubleTap))) { 
+         $0.numberOfTapsRequired = 2
+         $0.require(toFail: $0)
+         self.previewView.addGestureRecognizer($0)
+      }()
       with(UIPinchGestureRecognizer(target: self, action: #selector(onPreviewViewPinch))) {
          $0.delegate = self
          self.previewView.addGestureRecognizer( $0 )
       }
+      with(UILongPressGestureRecognizer(target: self, action: #selector(onPreviewViewLongPress))) {
+         self.previewView.addGestureRecognizer( $0 )
+      }
    }
    /**
     * Normal tap
     */
    @objc open func onPreviewViewTap(sender: UIGestureRecognizer) {
       guard [.ended, .cancelled, .failed].contains(sender.state) else { return }/*ensures that the tap isnt a swipe tap etc*/
       let devicePoint: CGPoint = (self.previewView.previewLayer).captureDevicePointConverted(fromLayerPoint: sender.location(in: sender.view))
-      Swift.print( devicePoint )
+      print("HybridCamera / CamView+Gesture / devicePoint:", devicePoint)
       self.focusWithMode(focusMode: .continuousAutoFocus, exposureMode: .continuousAutoExposure, point: devicePoint, monitorSubjectAreaChange: true)
    }
+    /**
+     * blurtime: Double tap --> Flip camera
+     */
+    @objc open func onPreviewViewDoubleTap(sender: UIGestureRecognizer) {
+        setCamera(cameraType: .back) // flipButton and toggle needed here...
+        print("double tap")
+    }
+
```

Comment:
I recommend installing swift-lint. It will tell you when syntax doesn't adhere to project .lint style guide.

Reply:
I'll check it out very soon, thank you! I seriously admire your code for being so crazy clean 😄